### ISSUE TITLE
Version pinning being ignored in conda 4.3?

### ISSUE THREAD

**Issue by mingwandroid** [conda/conda#4089]
Hi @kalefranz, @msarahan, I've run into either a bug in conda 4.3 on an incompatibility between conda and conda-build.

I am trying to rebuild Qt using the latest code from git for each, and Qt has quite detailed version pinning, yet, adding --debug to conda-build, I see:

```
Solving package specifications: DEBUG conda.resolve:solve(862): Solving for [MatchSpec('bison'), MatchSpec('dbus'), MatchSpec('flex'), MatchSpec('fontconfig'), MatchSpec('freetype'), MatchSpec('gcc'), MatchSpec('gst-plugins-base'), MatchSpec('icu'), MatchSpec('jpeg'), MatchSpec('libpng'), MatchSpec('libxcb'), MatchSpec('openssl'), MatchSpec('python'), MatchSpec('xz'), MatchSpec('zlib')]
```

.. (i.e. not a single version in those MatchSpecs) and the wrong python gets installed (3.5.2 instead of 2.7.12).

Reverting back to conda 4.2.13 I see:

```
Solving package specifications: .DEBUG conda.resolve:get_dists(383): Retrieving packages for: [MatchSpec('bison'), MatchSpec('dbus'), MatchSpec('flex'), MatchSpec('fontconfig 2.12.*'), MatchSpec('freetype  2.5*'), MatchSpec('gcc'), MatchSpec('gst-plugins-base'), MatchSpec('icu 54.*'), MatchSpec('jpeg 8*'), MatchSpec('libpng >=1.6.21,<1.7'), MatchSpec('libxcb'), MatchSpec('openssl 1.0*'), MatchSpec('python 2.7.*'), MatchSpec('xz'), MatchSpec('zlib 1.2*')]
```

.. and the right python gets installed.

**Comment by kalefranz** [comment]
> and the wrong python gets installed (3.5.2 instead of 2.7.12)

I know this part is fixed in the 4.3.x branch now.  It's possible Sophia also fixed the version pinning issue you're having too, if you want to give it a go before I'm able to try to reproduce it.

**Comment by mingwandroid** [comment]
> the wrong python gets installed

.. was the same problem as the version pinning I think? It looks to be fine now after an update. Thanks @kalefranz and @soapy1.
